src/compute/src/sink/persist_sink.rs

```diff
@@ -606,6 +609,21 @@ where
                     return;
                 }
 
+                // Make sure that our write handle lease does not expire!
+                //
+                // We don't write to _Consensus_ but instead only write batch
+                // data to _Blob_ so someone might eventually expire our lease.
+                // This could lead to the "leaked blob reaper" eventually
+                // removing blobs that have been written but not yet appended to
+                // the shard.
+                //
+                write.maybe_heartbeat_writer().await;
+                trace!("persist_sink {sink_id}/{shard_id}: writer heartbeating write handle");
+                // NOTE: We schedule ourselves to make sure that we keep
+                // heartbeating even in cases where there are no changes in our
+                // inputs (updates or frontier changes).
+                activator.activate_after(Duration::from_secs(60));
```

Comment:
does this do anything? if we have nothing to do and loop around to `scheduler.notified().await`, and we re-activate after 60 seconds, all that happens is that the `notified` future gets polled again and returns `Pending`

it seems to me that what we want is to run code every x seconds EVEN IF we aren't notified of any changes, which likely requires the new async integration, which will be hard to use here as `AsyncInputHandle` doesn't really have a good `for_each` parallel (we can probably add this)

Reply:
It does, yes. If you turn the logs into `info!` logging and the Duration to, say, `from_millis(10)` you will be flooded with logs. With `from_secs(60)` it's lost in the noise at steady state, with our 1sec timestamping interval.

Reply:
interesting, i need to read the `build_async` impl more closely